### PR TITLE
os/bluestore: simplified allocator interfaces to single apis

### PR DESCRIPTION
**Pull request by chhabaramesh**
Signed-off-by: Ramesh Chander <Ramesh.Chander@sandisk.com>

### CODE REVIEW

**Comment by chhabaramesh** [comment]
@liewegas please have a look at it. While I am doing more testing on it individually, maybe we can include this in normal runs ?

**Comment by chhabaramesh** [comment]
@liewegas , this one passes the ceph_objectstore test. Please review and merge.

File: src/os/bluestore/BitMapAllocator.cc [hunk 1]
```diff
@@ -182,26 +182,28 @@ int BitMapAllocator::alloc_extents_cont(
...
-  int64_t max_blks = max_alloc_size / m_block_size;
+  int64_t cont_blks = alloc_unit / m_block_size;
 
   ExtentList block_list = ExtentList(extents, m_block_size, max_alloc_size);
 
   while (need_blks > 0) {
     int64_t count = 0;
-    count = m_bit_alloc->alloc_blocks_res(
-      (max_blks && need_blks > max_blks) ? max_blks : need_blks, hint, &start_blk);
+    count = m_bit_alloc->alloc_blocks_res(cont_blks, hint, &start_blk);
```
Comment (liewegas) [inline src/os/bluestore/BitMapAllocator.cc]: This looks really inefficient.. if I'm reading this right, you're calling into alloc_blocks_res for *every* alloc_unit (min_alloc_size).  Is that right?

Comment (chhabaramesh) [inline src/os/bluestore/BitMapAllocator.cc]: @liewegas , yes you are right, I keep forgetting that min_alloc_size could change and we may fall in to this case more often than I think. 

I am modifying the existing alloc_extent code to search the free extents when min_alloc_size is not equal to bit in the bitmap.

Will update you soon.

**Pull request merged**